core/src/test/java/kafka/server/share/ShareFetchUtilsTest.java

```diff
@@ -300,4 +293,69 @@ public void testProcessFetchResponseWithLsoMovementForTopicPartition() {
+            groupId, Uuid.randomUuid().toString(), new CompletableFuture<>(), partitionMaxBytes);
+
+        ReplicaManager replicaManager = mock(ReplicaManager.class);
+
+        // Mock the replicaManager.fetchOffsetForTimestamp method to return a timestamp and offset for the topic partition.
+        FileRecords.TimestampAndOffset timestampAndOffset = new FileRecords.TimestampAndOffset(100L, 1L, Optional.empty());
+        doReturn(new OffsetResultHolder(Option.apply(timestampAndOffset), Option.empty())).when(replicaManager).fetchOffsetForTimestamp(any(TopicPartition.class), anyLong(), any(), any(), anyBoolean());
+
+        when(sp0.acquire(anyString(), any(FetchPartitionData.class))).thenReturn(Collections.emptyList());
+        doNothing().when(sp0).updateCacheAndOffsets(any(Long.class));
+
+        MemoryRecords records = MemoryRecords.withRecords(Compression.NONE,
+            new SimpleRecord("0".getBytes(), "v".getBytes()),
+            new SimpleRecord("1".getBytes(), "v".getBytes()),
+            new SimpleRecord("2".getBytes(), "v".getBytes()),
+            new SimpleRecord(null, "value".getBytes()));
+
+        // When no records are acquired from share partition.
+        Map<TopicIdPartition, FetchPartitionData> responseData = Collections.singletonMap(
+            tp0, new FetchPartitionData(Errors.NONE, 0L, 0L,
+                records, Optional.empty(), OptionalLong.empty(), Optional.empty(),
+                OptionalInt.empty(), false));
+
+        Map<TopicIdPartition, ShareFetchResponseData.PartitionData> resultData =
+            ShareFetchUtils.processFetchResponse(shareFetchData, responseData, sharePartitionManager, replicaManager);
+
+        assertEquals(1, resultData.size());
+        assertTrue(resultData.containsKey(tp0));
+        assertEquals(0, resultData.get(tp0).partitionIndex());
+        assertNull(resultData.get(tp0).records());
+        assertTrue(resultData.get(tp0).acquiredRecords().isEmpty());
+        assertEquals(Errors.NONE.code(), resultData.get(tp0).errorCode());
+
+        // When fetch partition data has OFFSET_OUT_OF_RANGE error.
+        responseData = Collections.singletonMap(
+            tp0, new FetchPartitionData(Errors.OFFSET_OUT_OF_RANGE, 0L, 0L,
+                records, Optional.empty(), OptionalLong.empty(), Optional.empty(),
+                OptionalInt.empty(), false));
+
+        resultData = ShareFetchUtils.processFetchResponse(shareFetchData, responseData, sharePartitionManager, replicaManager);
+
+        assertEquals(1, resultData.size());
+        assertTrue(resultData.containsKey(tp0));
+        assertEquals(0, resultData.get(tp0).partitionIndex());
+        assertNull(resultData.get(tp0).records());
+        assertTrue(resultData.get(tp0).acquiredRecords().isEmpty());
+        assertEquals(Errors.NONE.code(), resultData.get(tp0).errorCode());
+
+        Mockito.verify(sp0, times(1)).updateCacheAndOffsets(any(Long.class));
```

Comment:
nit: `fetchOffsetForTimestamp` is mock, so we can verify the actual value `1` instead of `any(Long.class)`

Reply:
Done.